### ISSUE TITLE
Graph HyperNetworks for Neural Architecture Search

### ISSUE THREAD

**Issue by sile** [sile/readings#78]
URL
-----

https://arxiv.org/abs/1810.05749

Abstract
----------

> Neural architecture search (NAS) automatically ﬁnds the best task-speciﬁc neural network topology, outperforming many manual architecture designs. However, it can be prohibitively expensive as the search requires training thousands of different networks, while each can last for hours. In this work, we propose the Graph HyperNetwork (GHN) to amortize the search cost: given an architecture, it directly generates the weights by running inference on a graph neural network. GHNs model the topology of an architecture and therefore can predict network performance more accurately than regular hypernetworks and premature early stopping. To perform NAS, we randomly sample architectures and use the validation accuracy of networks with GHN generated weights as the surrogate search signal. GHNs are fast – they can search nearly 10× faster than other random search methods on CIFAR-10 and ImageNet. GHNs can be further extended to the anytime prediction setting, where they have found networks with better speed-accuracy tradeoff than the state-of-the-art manual designs.

Table of Contents
--------------------

- (1) Introduction
- (2) Related Work
- (3) Background
  - Graph Neural Network
  - Hypernetwork
- (4) Graph Hypernetworks for Neural Architectural Search
  - (4-1) Graphical Representation
  - (4-2) Graph Hypernetwork
  - (4-3) Architectural Motifs and Stacked GNNs
  - (4-4) Forward-backward GNN message passing
  - (4-5) Learning
- (5) Experiments
  - (5-1) NAS Benchmarks
    - (5-1-1) CIFAR-10
    - (5-1-2) ImageNet-Mobile
  - (5-2) Anytime Prediction
  - (5-3) Predicted Performance Correlation (CIFAR-10)
  - (5-4) Ablation Studies (CIFAR-10)
    - Number of graph nodes
    - Number of propagation steps
    - Stacked GHN for architectural motifs
- (6) Conclusion

BibTeX
------

```bibtex
@article{zhang2018graph,
  title={Graph hypernetworks for neural architecture search},
  author={Zhang, Chris and Ren, Mengye and Urtasun, Raquel},
  journal={arXiv preprint arXiv:1810.05749},
  year={2018}
}
```
